### DISCUSSION TITLE
Feature Request: Create Static Routes by configuration.

### DISCUSSION THREAD

**Discussion by Visolo** [fabianishere/udm-iptv#206] [Ideas]
Hi

Sorry for my english.

This is a feature request.

Mi ISP (Vodafone Spain Direct FFTH - No NEBA) dont send data in dhcp. I have to use IPTV_WAN_DHCP_OPTIONS="-n -o".

When I do this, IPTV is working fine, but records and other services are not working. Searching in my ISP Router I have find that there are some static routes in the configuration.

I have solved this adding them manualy from ssh console with this code:
```
GW=$(route -n | awk '/iptv/ { print $1 }' | sed 's/.$/1/')
ip route add 10.179.32.0/23 via $GW
ip route add 10.15.220.0/24 via $GW
ip route add 110.8.59.0/24 via $GW
ip route add 10.8.58.0/24 via $GW
ip route add 10.8.57.0/24 via $GW
```

These ranges are in my ISP Router. I Dont konw if are diferent in another client.
But this is a bullshit, becouse in every restart I have to do it again.

Now I have made a litte modification in you code.
1- Add a option in config file

```
# Static Routes if needed
IPTV_STATIC_ROUTES="10.179.32.0/23 10.15.220.0/24 110.8.59.0/24 10.8.58.0/24 10.8.57.0/24"
```

2- Add code /usr/lib/udm-iptv/udm-iptv in _network_setup() just after "NAT the IPTV ranges"

```
    echo "Creating static routes (if necesary)"
    # Create static routes if needed
    GW=$(route -n | awk '/iptv/ { print $1 }' | sed 's/.$/1/')
    for range in $IPTV_STATIC_ROUTES; do
        ip route add "$range" via $GW
    done
```

This works as expected and in every restart the routes are created correctly and all work fine.

¿Can you add this to your code, so in the next update i will not need to add again?

Thanks a lot.

**Comment by dogstyx** [comment]
I concur with this request, I see that the static routes functionality has been added in the 3.0.0 version but I would need the routes to be added through the gateway in the above mentioned fashion, otherwise it doesn't work for me.

**Reply by Visolo** [reply]
Yes, its the same for me.
I have to patch the code in every release. 
I need for my operator that static routes be added through the gateway.
